### DISCUSSION TITLE
Get compiled css from content editor in production

### DISCUSSION THREAD

**Discussion by 2acode** [tailwindlabs/tailwindcss#9994] [Help]
Hi,
the user creates the page via editor and saves on the database.
On the frontend it is processed and injected into the layout.

Since it's specified on the site that the PlayCDN shouldn't be used in production, what's the best way to get the CSS compiled from a code editor on a production site?

An API with the same logic as PlayCDN (JIT) that returns the compiled CSS to be saved and then linked to the page in the frontend.

Can I take advantage of the PlayCDN or is there a dedicated service?
It can be done with javascript on the production server (project based on Laravel, I can still install node if needed).

Thank you

**Comment by wongjn** [comment]
You could use your own custom solution that runs Tailwind via PostCSS's JavaScript API:
```js
import postcss from 'postcss';
import tailwindcss from 'tailwindcss';
import base from './tailwind.config';

const sourceCSS = '@tailwindcss base; @tailwindcss components; @tailwind utilities';
const config = {
  presets: [base],
  content: [{ raw: getPage(), extension: 'html' }],
};
const css = await postcss([tailwindcss(config)]).process(sourceCSS);
